src/dispatching2/dialogue/dialogue_handler_ext.rs

```diff
@@ -0,0 +1,34 @@
+use crate::dispatching2::dialogue::{get_chat_id::GetChatId, Dialogue, Storage};
+use dptree::{di::DependencyMap, Handler};
+use std::sync::Arc;
+
+pub trait DialogueHandlerExt {
+    #[must_use]
+    fn add_dialogue<Upd, S, D>(self) -> Self
```

Comment:
Same as with `add_commands` - shouldn't this be called `filter_dialogue`?

Reply:
Also, can't this be part of the `HandlerExt`?

Reply:
Fixed.